.github/workflows/ci.yml

```diff
@@ -48,15 +52,31 @@ jobs:
       - name: Run cargo release build
         run: cargo build --release
 
-  # Enable these tests once there's a runner with a GPU.
-  #test_gpu:
-  #  runs-on: ubuntu-24.04
-  #  name: Test
-  #  steps:
-  #    - uses: actions/checkout@v4
-  #    - name: Install required packages
-  #      run: sudo apt install --no-install-recommends --yes libhwloc-dev nvidia-cuda-toolkit ocl-icd-opencl-dev
-  #    - name: Run tests
-  #      run: cargo test
-  #    - name: Run `add` example
-  #      run: cargo run --example add
+  test_gpu:
+    runs-on: ['self-hosted', 'linux', 'x64', '2xlarge+gpu']
+    name: Test
+    steps:
+      - uses: actions/checkout@v4
+      # TODO: Move the driver installation to the AMI.
+      # https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/install-nvidia-driver.html
+      # https://www.nvidia.com/en-us/drivers/
+      - name: Install CUDA drivers
+        run: |
+          curl -L -o nvidia-driver-local-repo-ubuntu2404-570.148.08_1.0-1_amd64.deb https://us.download.nvidia.com/tesla/570.148.08/nvidia-driver-local-repo-ubuntu2404-570.148.08_1.0-1_amd64.deb
+          sudo dpkg -i nvidia-driver-local-repo-ubuntu2404-570.148.08_1.0-1_amd64.deb
+          sudo cp /var/nvidia-driver-local-repo-ubuntu2404-570.148.08/nvidia-driver-local-*-keyring.gpg /usr/share/keyrings/
+          sudo apt-get update
+          sudo apt-get install --no-install-recommends --yes cuda-drivers
+          rm nvidia-driver-local-repo-ubuntu2404-570.148.08_1.0-1_amd64.deb
+      - name: Install required packages
+        run: |
+          sudo apt-get update
+          sudo apt-get install --no-install-recommends --yes libhwloc-dev nvidia-cuda-toolkit ocl-icd-opencl-dev
+      # TODO: Remove this and other rust installation directives from jobs running
+      - uses: dtolnay/rust-toolchain@21dc36fb71dd22e3317045c0c31a3f4249868b17
+        with:
+          toolchain: 1.83
```

Comment:
Note that this repo is on Rust 1.81. (see the `rust-toolchain`) file. Hence it makes sense to have `rustup` properly installed on the runner, so you don't have to care about it in the CI file.